runtime/v2/shim/util_windows.go

```diff
@@ -48,7 +49,24 @@ func SocketAddress(ctx context.Context, id string) (string, error) {
 
 // AnonDialer returns a dialer for a npipe
 func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
-	return winio.DialPipe(address, &timeout)
+	var c net.Conn
+	var lastError error
+	start := time.Now()
+	for {
+		remaining := timeout - time.Now().Sub(start)
+		if remaining <= 0 {
+			lastError = errors.Errorf("timed out waiting for npipe %s", address)
+			break
+		}
+		c, lastError = winio.DialPipe(address, &remaining)
```

Comment:
if it does not exist, does it return immediately or does it use the timeout?

Reply:
@crosbymichael - Hang on this is the wrong commit. I dont quite know what happened but this doesnt have the right exit logic it it connects successfully.

Reply:
I mean if you are still working on it, you could maybe use a:
```go
select {
case <-time.After(timeout):
case <-pipe:
...
}
````

as a way to synchronize

Reply:
Ok this is now right.

I can do it that way if you prefer. Doesn't bother me either way. Since winio returns immediately for file not found it would still require the same looping logic and starting a new select query so it seems the same code in the end.